pkg/dmlegacy/fstab.go

```diff
@@ -0,0 +1,119 @@
+package dmlegacy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"text/tabwriter"
+
+	api "github.com/weaveworks/ignite/pkg/apis/ignite"
+)
+
+const (
+	mountOptions = "rw,relatime"
+	uuidPath     = "/dev/disk/by-uuid"
+)
+
+type fstabEntry struct {
+	uuid       string
+	mountPoint string
+}
+
+var _ fmt.Stringer = &fstabEntry{}
+
+func (f *fstabEntry) isValid() bool {
+	// An entry is valid if both the UUID and mount point are set
+	return len(f.uuid) > 0 && len(f.mountPoint) > 0
+}
+
+func (f *fstabEntry) String() string {
+	return strings.Join([]string{
+		fmt.Sprintf("UUID=%s", f.uuid), // Mount by UUID
+		f.mountPoint,                   // The mount point for the volume
+		"auto",                         // Discover the filesystem automatically
+		mountOptions,                   // Use the mount options defined above
+		"0",                            // Don't dump the filesystem
+		"2",                            // fsck may check this filesystem on reboot
+	}, "\t")
+}
+
+func populateFstab(vm *api.VM, mountPoint string) error {
+	fstab, err := os.Create(path.Join(mountPoint, "/etc/fstab"))
+	if err != nil {
+		return err
+	}
+	defer fstab.Close()
+
+	writer := new(tabwriter.Writer)
+	writer.Init(fstab, 0, 8, 1, '\t', 0)
+	entries := make(map[string]*fstabEntry)
+
+	// Discover all volumes
+	for _, volume := range vm.Spec.Storage.Volumes {
+		if volume.BlockDevice == nil {
+			continue // Skip all non block device volumes for now
+		}
+
+		// Retrieve the UUID for the block device
+		uuid, err := getUUID(volume.BlockDevice.Path)
+		if err != nil {
+			return err
+		}
+
+		// Create an entry for the volume
+		entries[volume.Name] = &fstabEntry{uuid: uuid}
+	}
+
+	// Discover all volume mounts
+	for _, volumeMount := range vm.Spec.Storage.VolumeMounts {
+		// Lookup the entry based on the volume mount's name
+		if entry, ok := entries[volumeMount.Name]; ok {
+			// Add the mount path to the entry if it exists
+			entry.mountPoint = volumeMount.MountPath
+		}
+	}
+
+	for _, entry := range entries {
+		if entry.isValid() {
```

Comment:
at least put a warning if it's not valid? log.Errorf?

Reply:
It now returns an error in ca75db1, this should never happen thanks to validation.